jupyterhub/values.yaml

```diff
@@ -109,16 +109,22 @@ hub:
   templatePaths: []
   templateVars: {}
   livenessProbe:
+    # The livenessProbe's aim to give JupyterHub sufficient time to startup but
+    # be able to restart if it becomes unresponsive for ~5 min.
     enabled: false
-    initialDelaySeconds: 180
+    initialDelaySeconds: 300
     periodSeconds: 10
-    failureThreshold: 3
-    timeoutSeconds: 1
+    failureThreshold: 30
+    timeoutSeconds: 3
   readinessProbe:
+    # The readinessProbe's aim is to provide a successful startup indication,
+    # but following that never become unready before its livenessProbe fail and
+    # restarts it if needed. To become unready following startup serves no
+    # purpose as there are no other pod to fallback to in our non-HA deployment.
     enabled: true
     initialDelaySeconds: 0
     periodSeconds: 2
```

Comment:
Could change this to e.g. 10 since there's no need for such frequent checks? Same for chp.

Reply:
The compromise is:

- On the one hand, make hub/proxy pod respond quickly following start to external web requests.
- On the other hand, avoid spamming the hub/proxy pod.

Note also the interaction between the hub/proxy pod. On startup the proxy pod is being configured by REST calls from the hub pod, at least assuming it is k8s Ready because otherwise it won't receive network traffic. This makes a delay in the proxy pods readiness risk propegate to slow down the hub's initial startup procedure as well.

I intentionally considered this compromise in the past, and still think 2 seconds is a sensible compromise. We have made the requests to these endpoint not be logged by default to avoid cluttering logs in the latest version of JupyterHub, perhaps also CHP? I don't remember, if not, I think we should.

Reply:
Fair enough 😄 . We can always change it later since this is the existing value